examples/userapi/subdomains.ipynb

```diff
@@ -0,0 +1,966 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 22,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "class Right(SubDomain):\n",
+    "    name = 'right'\n",
+    "    def define(self, dimensions):\n",
+    "        x, y = dimensions\n",
+    "        return {x: x, y: ('right', 2)}\n",
+    "rd = Right()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Note that owing to the chosen definitions of 'left' and 'right' there will be four grid points on which these two subdomains overlap.\n",
+    "\n",
+    "Then, let us create a grid consisting of the subdomains 'middle', 'left' and 'right' and specify an equation on each of the three subdomains:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 23,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "grid = Grid(shape = (10, 10), subdomains = (mid, ld, rd))\n",
+    "f = Function(name = 'f', grid = grid)\n",
+    "\n",
+    "eq1 = Eq(f, f+1, subdomain = grid.subdomains['middle'])\n",
+    "eq2 = Eq(f, f+2, subdomain = grid.subdomains['left'])\n",
+    "eq3 = Eq(f, f+3, subdomain = grid.subdomains['right'])"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We then create and execute an operator that evaluates the above equations:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 24,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "Operator `Kernel` run in 0.01 s\n"
+     ]
+    }
+   ],
+   "source": [
+    "#NBVAL_IGNORE_OUTPUT\n",
+    "op2 = Operator([eq1, eq2, eq3])()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Viewing the data of `f` after performing the operation we see that:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 25,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "Data([[2., 2., 2., 2., 2., 2., 2., 2., 5., 5.],\n",
+       "      [2., 2., 2., 2., 2., 2., 2., 2., 5., 5.],\n",
+       "      [0., 0., 0., 0., 0., 0., 0., 0., 3., 3.],\n",
+       "      [0., 0., 0., 0., 1., 1., 1., 0., 3., 3.],\n",
+       "      [0., 0., 0., 0., 1., 1., 1., 0., 3., 3.],\n",
+       "      [0., 0., 0., 0., 1., 1., 1., 0., 3., 3.],\n",
+       "      [0., 0., 0., 0., 0., 0., 0., 0., 3., 3.],\n",
+       "      [0., 0., 0., 0., 0., 0., 0., 0., 3., 3.],\n",
+       "      [0., 0., 0., 0., 0., 0., 0., 0., 3., 3.],\n",
+       "      [0., 0., 0., 0., 0., 0., 0., 0., 3., 3.]], dtype=float32)"
+      ]
+     },
+     "execution_count": 25,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "f.data"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Seismic wave propagation example with subdomains\n",
```

Comment:
I wonder whether this should be moved to a different notebook somewhere under `examples/seismic` ? clearly stating then that `userapi/subdomains.ipynb` is a prerequisite 

Thoughts?

Reply:
I prefer keeping it here since after going through the basics I think it's nice to directly have a more 'practical' example.